examples/platform_channel/lib/main.dart

```diff
@@ -13,36 +13,67 @@ class PlatformChannel extends StatefulWidget {
 }
 
 class _PlatformChannelState extends State<PlatformChannel> {
-  static const PlatformMethodChannel platform = const PlatformMethodChannel('samples.flutter.io/battery');
-  String _batteryLevel = '';
+  static const PlatformMethodChannel methodChannel =
+      const PlatformMethodChannel('io.flutter.samples/battery');
+  static const PlatformEventChannel eventChannel =
+      const PlatformEventChannel('io.flutter.samples/charging');
+
+  String _batteryLevel = 'Battery level: unknown.';
+  String _chargingStatus = 'Battery status: unknown.';
 
   Future<Null> _getBatteryLevel() async {
     String batteryLevel;
     try {
-      final int result = await platform.invokeMethod('getBatteryLevel');
-      batteryLevel = 'Battery level at $result % .';
-    } on PlatformException catch (e) {
-      batteryLevel = "Failed to get battery level: '${e.message}'.";
+      final int result = await methodChannel.invokeMethod('getBatteryLevel');
+      batteryLevel = 'Battery level: $result%.';
+    } on PlatformException {
+      batteryLevel = "Failed to get battery level.";
     }
     setState(() {
       _batteryLevel = batteryLevel;
     });
   }
 
   @override
+  void initState() {
+    super.initState();
+    eventChannel.receiveBroadcastStream().listen(_onEvent, onError: _onError);
+  }
+
+  void _onEvent(String event) {
+    setState(() {
+      _chargingStatus =
+          "Battery status: ${event == 'charging' ? '' : 'dis'}charging.";
+    });
+  }
+
+  void _onError(PlatformException error) {
+    setState(() {
+      _chargingStatus = "Battery status: unknown.";
+    });
+  }
+
+  @override
   Widget build(BuildContext context) {
     return new Material(
-      child: new Center(
-        child: new Column(
-          mainAxisAlignment: MainAxisAlignment.spaceEvenly,
-          children: <Widget>[
-            new RaisedButton(
-              child: new Text('Get Battery Level'),
-              onPressed: _getBatteryLevel,
-            ),
-            new Text(_batteryLevel, key: new Key('Battery level label')),
-          ],
-        ),
+      child: new Column(
+        mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+        children: <Widget>[
+          new Column(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: <Widget>[
+              new Text(_batteryLevel, key: new Key('Battery level label')),
+              new Padding(
+                padding: new EdgeInsets.all(15.0),
```

Comment:
I think material recommends a grid based on whole multiple of 8dp: https://material.io/guidelines/layout/metrics-keylines.html